Source/OCMock/OCMockObject.m

```diff
@@ -31,15 +31,80 @@
+
+// gCurrentMocksToStopRecorder will point to either gTestCaseMocksToStop or gTestSuiteMocksToStop
+// depending on whether we are currently recoring testcase mocks or test suite mocks.
+// Controlled by OCMockXCTestObserver.
+static NSHashTable<OCMockObject *> *gTestCaseMocksToStop;
+static NSHashTable<OCMockObject *> *gTestSuiteMocksToStop;
+static NSHashTable<OCMockObject *> *gCurrentMocksToStopRecorder;
+
+// Flag that controls whether we should be asserting after stopmocking is called.
+// Controlled by OCMockXCTestObserver.
+static BOOL gAssertOnCallsAfterStopMocking;
+
+// "Fake" Protocol so we can avoid having to link to XCTest, but not get warnings about
+// methods not being declared.
+@protocol OCMockXCTestObservation
++ (id)sharedTestObservationCenter;
+- (void)addTestObserver:(id)observer;
+@end
 
-@implementation OCMockObject
+@interface OCMockXCTestObserver : NSObject
+@end
 
+@implementation OCMockObject
 #pragma mark  Class initialisation
++ (void)load
+{
+    gTestCaseMocksToStop = [[NSHashTable weakObjectsHashTable] retain];
+    gTestSuiteMocksToStop = [[NSHashTable weakObjectsHashTable] retain];
+    gCurrentMocksToStopRecorder = nil;
+    gAssertOnCallsAfterStopMocking = YES;
+    Class xctest = NSClassFromString(@"XCTestObservationCenter");
+    if (xctest)
+    {
+        // If XCTest is available, we set up an observer to stop our mocks for us.
+        [[xctest sharedTestObservationCenter] addTestObserver:[[OCMockXCTestObserver alloc] init]];
+    }
+
+}
 
 + (void)initialize
 {
-	if([[NSInvocation class] instanceMethodSignatureForSelector:@selector(getArgumentAtIndexAsObject:)] == NULL)
-		[NSException raise:NSInternalInconsistencyException format:@"** Expected method not present; the method getArgumentAtIndexAsObject: is not implemented by NSInvocation. If you see this exception it is likely that you are using the static library version of OCMock and your project is not configured correctly to load categories from static libraries. Did you forget to add the -ObjC linker flag?"];
+    if (self == [OCMockObject class]) {
+        if([[NSInvocation class] instanceMethodSignatureForSelector:@selector(getArgumentAtIndexAsObject:)] == NULL)
+        {
+            [NSException raise:NSInternalInconsistencyException format:@"** Expected method not present; the method getArgumentAtIndexAsObject: is not implemented by NSInvocation. If you see this exception it is likely that you are using the static library version of OCMock and your project is not configured correctly to load categories from static libraries. Did you forget to add the -ObjC linker flag?"];
+        }
+    }
+}
+
+#pragma mark  Mock cleanup recording
+
++ (void)recordAMockToStop:(OCMockObject *)mock {
+  [gCurrentMocksToStopRecorder addObject:mock];
+}
+
++ (void)removeAMockToStop:(OCMockObject *)mock {
+  [gCurrentMocksToStopRecorder removeObject:mock];
+}
+
++ (void)stopAllTestCaseMocks {
+    for (OCMockObject *mock in gTestCaseMocksToStop)
```

Comment:
This needs to be customizable. It's not unusual for clients with high concurrency to clean up mocks after the tests have finished.